### PR TITLE
feat(MdnCommand): add mdn search command

### PR DESCRIPTION
**Pull request by Beatzoid**
Add a JSDoc command which allows you to search the MDN Web Docs for terms. Example: https://imgur.com/a/RZgClP7

### CODE REVIEW

**Comment by Beatzoid** [comment]
Not ready to merge until Milo comes out with a custom pagination library that has Types

**Comment by Milo123459** [comment]
Follow updates [here](https://github.com/Milo123459/ducki)

File: src/commands/Info Commands/JsDocCommand.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+				return message.channel.send(
+					client.embed({ title: 'No results!' }, message)
+				);
+			}
+
+			paginationEmbed(message, pages, ['⏪', '⏩'], 30000);
+		});
+};
+
+export const name: string = 'jsdoc';
```
Comment (Milo123459) [inline src/commands/Info Commands/JsDocCommand.ts:45]: name it mdn, change file name as well

File: src/commands/Info Commands/MdnCommand.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { MessageEmbed } from 'discord.js';
+import nodeFetch from 'node-fetch';
+import { RunFunction } from '../../interfaces/Command';
+import paginationEmbed from 'discord.js-pagination';
+
+export const run: RunFunction = async (client, message, args) => {
+	if (!args.length)
+		return message.channel.send(
```
Comment (Milo123459) [inline src/commands/Info Commands/MdnCommand.ts:8]: please use client.embed for an error message

File: src/commands/Info Commands/MdnCommand.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	)
+		.then((body) => body.json())
+		.then(async (result) => {
+			result.documents.forEach((document) => {
+				if (!document.slug.toLowerCase().includes('javascript')) {
+					return;
+				}
+
+				pages.push(
+					new MessageEmbed()
```
Comment (Milo123459) [inline src/commands/Info Commands/MdnCommand.ts:27]: use client.embed

File: src/commands/Info Commands/MdnCommand.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { RunFunction } from '../../interfaces/Command';
+import paginationEmbed from 'discord.js-pagination';
+
+export const run: RunFunction = async (client, message, args) => {
+	if (!args.length)
+		return message.channel.send(
+			'Invalid usage, provide a search term! `mdn <search input>`'
+		);
+
+	const pages = [];
```
Comment (Milo123459) [inline src/commands/Info Commands/MdnCommand.ts:12]: make this an array of MessageEmbed

File: src/commands/Info Commands/MdnCommand.ts (hunk 3 above)
Comment (Beatzoid) [reply src/commands/Info Commands/MdnCommand.ts:27]: I can't use client.embed for the pages because the pagination package sets its own footers

Comment (Milo123459) [reply src/commands/Info Commands/MdnCommand.ts:30]: OK

File: src/utils/Pagination.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+				page = page > 0 ? --page : pages.length - 1;
+				break;
+			case emojiList[1]:
+				page = page + 1 < pages.length ? ++page : 0;
+				break;
+			default:
+				break;
+		}
+
+		curPage.edit(pages[page].setFooter(`Page ${page + 1} / ${pages.length}`));
```
Comment (Milo123459) [inline src/utils/Pagination.ts:37]: use author instead of footer

**Review by Milo123459 — APPROVED** [review]

**Pull request merged**